lib/Doctrine/DBAL/Platforms/AbstractPlatform.php

```diff
@@ -2065,6 +2065,11 @@ protected function getPreAlterTableIndexForeignKeySQL(TableDiff $diff)
             $sql[] = $this->getDropIndexSQL($index, $tableName);
         }
         foreach ($diff->changedIndexes as $index) {
+            if ($index->isPrimary()) {
+                $sql[] = $this->getDropPrimaryKeySQL($tableName);
```

Comment:
@Majkl578 is there a way to enforce a new line before flow control operators like `return`, `continue`, etc on the code sniffer level? It would be very nice to have.

Reply:
No sniff I'm aware of, but shouldn't be hard to implement. Just needs to be compatible i.e. with single statements:
```php
if (...) {
    continue;
}
```
or comments:
```php
foo();

// comment
return 123;
```

Reply:
Yeah. More generically the first case is if the previous line has flow control statement which is a block opener, there shouldn't be an empty line. Like:
```php
function x()
{
    foreach (...) {
        foreach (...) {
        }
    }

    foreach (...) {
    }
}
```

Reply:
I've [added new line](https://github.com/doctrine/dbal/pull/2929/files#diff-3ee36db88cb7fcfb3df4415f6952fcd0R2083) before `continue`.

Reply:
@carusogabriel do you want to file this as a requirement for coding standards (I see we have disabled issues for the repository). It'd be nice to have a reference point somewhere. Otherwise, I constantly forget where the discussion started.

/cc @deeky666

Reply:
@morozov Require new lines before control structions you mean? :)

Reply:
Yes. Before and after.

Reply:
@morozov , Totally agree with this proposal, but code marked as still required your review :) Mark it properly please.

Reply:
@SCIF it's not because of the formatting issue. I want to see if the behavior of dropping a non-existing PK is consistent with other platforms.

Reply:
This thread is regarding cs only, the PK ones [here](https://github.com/doctrine/dbal/pull/2929#discussion_r178398463)